ext/snmp/snmp.c

```diff
@@ -829,31 +829,50 @@ static bool php_snmp_parse_oid(
 }
 /* }}} */
 
-/* {{{ netsnmp_session_init
-	allocates memory for session and session->peername, caller should free it manually using netsnmp_session_free() and efree()
+/* {{{ snmp_session_init
+	allocates memory for session and session->peername, caller should free it manually using session_free() and efree()
 */
-static bool netsnmp_session_init(php_snmp_session **session_p, int version, zend_string *hostname, zend_string *community, int timeout, int retries)
+static bool snmp_session_init(php_snmp_session **session_p, int version, zend_string *hostname, zend_string *community, zend_long timeout, zend_long retries)
 {
 	php_snmp_session *session;
 	char *pptr, *host_ptr;
 	bool force_ipv6 = false;
 	int n;
 	struct sockaddr **psal;
 	struct sockaddr **res;
+
+	*session_p = 0;
+
+	if (ZSTR_LEN(hostname) >= MAX_NAME_LEN) {
+		zend_value_error("hostname length must be lower than %d", MAX_NAME_LEN);
+		return false;
+	}
+
+	if (timeout < -1 || timeout > LONG_MAX) {
+		zend_value_error("timeout must be between -1 and %ld", LONG_MAX);
+		return false;
+	}
+
+	if (retries < -1 || retries > INT_MAX) {
+		zend_value_error("retries must be between -1 and %d", INT_MAX);
+		return false;
+	}
```

Comment:
Can we pass the arg_nums to these parameters? Does the community zend_string need to be checked too? Is Hostname checked to not contain null bytes prior to calling this function? If yes please add an assertion.